pysmt/numeral.py

```diff
@@ -0,0 +1,21 @@
+import warnings
+
+use_z3 = False
+try:
+    import z3num
+    use_z3 = True
+except:
+    pass
+
+if use_z3:
+    class Numeral(z3num.Numeral):
+        """Represents a Number (Algebraic)"""
+        def __hash__(self):
+            return hash(self.sexpr())
+
+else:
+    class Numeral(object):
+        """Represents a Number (Algebraic)"""
+        def __init__(self, obj):
+            self.obj = obj
+            warnings.warn("Z3 is not installed. We currently do not support stand-alone algebraic numbers. This might lead to unexpected results")
```

Comment:
It seemed too harsh... You should still be able to get a model, as long as you treat the object as an opaque object. As soon as you try to do anything on it, it will break. This should never happen right now, so I am fine with converting it into ```NotImplementedError```.